### PR TITLE
Export PDF's using jsPDF

### PR DESCRIPTION
**Pull request by bartvde**
This work is a follow-up to #2490 by @gingerik 
The example was updated to use the templates, and made some small modifications to the print button.

### CODE REVIEW

**Comment by marcjansen** [comment]
Back in #2490 we couldn't get an agreement on the added `@api` declarations. @elemoine wnated to look at it if I remember correctly.

**Comment by bartvde** [comment]
yeah I was also wondering if we could have an event on the tile queue so we don't need to poll it but not sure

**Comment by bartvde** [comment]
see https://github.com/bartvde/ol3/commit/a38721af36ece222a874bcd721180ca6fd057f54 for an alternative approach, this is using the generic change event (but this is only temporary to discuss this solution)

**Comment by bartvde** [comment]
I do have situations in which I get black tiles in the PDF, but I thought I saw the same with the previous approach if I am not mistaken

screen shot 2015-08-13 at 11 22 33

**Comment by ahocevar** [comment]
I may be missing something, but can't you use the 'loadend' event that we have available now?

**Comment by bartvde** [comment]
@ahocevar you mean ```tileloadend``` ?

**Comment by ahocevar** [comment]
@bartvde yes, 'tileloadend', sorry.

**Comment by bartvde** [comment]
@ahocevar good point, I think that can work as well (will push a commit soon). It leads to a bit more application code. The original PR was probably done before we had the load events. I am still seeing some black spots in the PDF as well in this case.

**Comment by gingerik** [comment]
Good to see this PR is being picked up again! I think the black tiles are caused by a delay between loading images and actually drawing them onto the canvas. Let me know when I can still help :-)

File: examples/export-pdf.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  map.once('postcompose', function(event) {
+    source.on('tileloadstart', function() {
+      ++loading;
+    });
+    source.on('tileloadend', function() {
+      ++loaded;
+      if (loading === loaded) {
+        loading = 0;
+        loaded = 0;
+      var canvas = event.context.canvas;
```
Comment (marcjansen) [inline examples/export-pdf.js]: Why didn't `gjslint` belch here and in the following lines?

Comment (bartvde) [inline examples/export-pdf.js]: Good point ;-) btw I still need to clean this up a bit.

**Comment by bartvde** [comment]
sorry for the wait but this is now ready for review cc @marcjansen

**Comment by marcjansen** [comment]
Sorry this slipped through, Bart. I do not have time for a big review, but it is 'just' an example… so: :+1: please merge.

**Comment by bartvde** [comment]
thanks @marcjansen

**Pull request merged**